File: tests/leap_compat/kotlin/PayloadConsumer.kt
package consumer

import ai.liquid.leap.function.LeapFunctionCall
import ai.liquid.leap.message.ChatMessage
import ai.liquid.leap.message.ChatMessageContent
import ai.liquid.leap.message.GenerationFinishReason
import ai.liquid.leap.message.GenerationStats
import ai.liquid.leap.message.MessageResponse

fun payloads(
    samples: FloatArray,
    error: Throwable,
): List<MessageResponse> {
    val call = LeapFunctionCall(name = "lookup", arguments = mapOf("key" to "value"))
    val message =
        ChatMessage(
            role = ChatMessage.Role.ASSISTANT,
            content = listOf(ChatMessageContent.Text("answer")),
            reasoningContent = "reason",
            functionCalls = listOf(call),
        )
    val stats = GenerationStats(promptTokens = 2, completionTokens = 1, totalTokens = 3, tokenPerSecond = 1f, cachedPromptTokens = 0)
    return listOf(
        MessageResponse.Chunk("answer"),
        MessageResponse.ReasoningChunk("reason"),
        MessageResponse.FunctionCalls(listOf(call)),
        MessageResponse.AudioSample(samples, 16000),
        MessageResponse.Error(error, "failed"),
        MessageResponse.Complete(message, GenerationFinishReason.STOP, stats),
    )
}

File: tests/leap_compat/swift/PayloadConsumer.swift
import LeapSDK

func payloads(samples: KotlinFloatArray, error: KotlinThrowable) -> [any MessageResponse] {
  let call = LeapFunctionCall(name: "lookup", arguments: ["key": "value"])
  let message = ChatMessage(
    role: .assistant, content: [ChatMessageContent.Text(text: "answer")],
    reasoningContent: "reason", functionCalls: [call])
  let stats = GenerationStats(
    promptTokens: 2, completionTokens: 1, totalTokens: 3,
    tokenPerSecond: 1, cachedPromptTokens: 0)
  let complete = MessageResponseComplete(fullMessage: message, finishReason: .stop, stats: stats)
  let _: [ChatMessageContent] = complete.fullMessage.content
  let _: Int64? = complete.stats?.cachedPromptTokens
  let _: String = call.name
  let _: [String: Any?] = call.arguments
  return [
    MessageResponseChunk(text: "answer"), MessageResponseReasoningChunk(reasoning: "reason"),
    MessageResponseFunctionCalls(functionCalls: [call]),
    MessageResponseAudioSample(samples: samples, sampleRate: 16000),
    MessageResponseError(throwable: error, message: "failed"), complete,
  ]
}
